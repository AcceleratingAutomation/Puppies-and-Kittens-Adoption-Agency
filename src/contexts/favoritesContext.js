import React, { createContext, useReducer } from 'react';

export const initialState = {
  adopters: [],
  fosters: [],
  rescues: [],
  veterinarians: [],
  users: [],
  rescueDetails: null,
  favorites: [],
  loading: true,
  openDialog: false
};

export function reducer(state, action) {
  switch (action.type) {
    case 'setAdopters':
      return { ...state, adopters: action.value, loading: false };
    case 'setFosters':
      return { ...state, fosters: action.value, loading: false };
    case 'setRescues':
      return { ...state, rescues: action.value, loading: false };
    case 'setUsers':
      return { ...state, users: action.value, loading: false };
    case 'setVeterinarians':
      return { ...state, veterinarians: action.value, loading: false };
    case 'setFavorites':
      return { ...state, favorites: action.value, loading: false };
    case 'addToFavorites':
      return {
        ...state,
        rescues: state.rescues.map(rescue =>
          rescue.id === action.id ? { ...rescue, isFavorite: true } : rescue
        ),
      };
    case 'removeFromFavorites':
      return {
        ...state,
        rescues: state.rescues.map(rescue =>
          rescue.id === action.id ? { ...rescue, isFavorite: false } : rescue
        ),
        favorites: state.favorites.filter(favoriteId => favoriteId !== action.id)
      };
    case 'setLoading':
      return { ...state, loading: action.value };
    default:
      throw new Error();
  }
}

export const FavoritesContext = createContext();

export const FavoritesProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <FavoritesContext.Provider value={{ state, dispatch }}>
      {children}
    </FavoritesContext.Provider>
  );
};