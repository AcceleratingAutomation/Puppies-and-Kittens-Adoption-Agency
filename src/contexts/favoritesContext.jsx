import React, { createContext, useReducer, useMemo } from "react";
import PropTypes from "prop-types";

export const initialState = {
  adopters: [],
  favorites: [],
  fosters: [],
  loading: true,
  openDialog: false,
  rescueDetails: null,
  rescues: [],
  veterinarians: [],
  users: [],
};

export function reducer(state, action) {
  switch (action.type) {
    case "setAdopters":
      return { ...state, adopters: action.value, loading: false };
    case "setFosters":
      return { ...state, fosters: action.value, loading: false };
    case "setRescues":
      return { ...state, rescues: action.value, loading: false };
    case "setRescueDetails":
      return { ...state, rescueDetails: action.value, loading: false };
    case "setUsers":
      return { ...state, users: action.value, loading: false };
    case "setVeterinarians":
      return { ...state, veterinarians: action.value, loading: false };
    case "setFavorites":
      return { ...state, favorites: action.value, loading: false };
    case "addToFavorites":
      return {
        ...state,
        rescues: state.rescues.map((rescue) =>
          rescue.id === action.id ? { ...rescue, isFavorite: true } : rescue,
        ),
      };
    case "removeFromFavorites":
      return {
        ...state,
        rescues: state.rescues.map((rescue) =>
          rescue.id === action.id ? { ...rescue, isFavorite: false } : rescue,
        ),
        favorites: state.favorites.filter(
          (favoriteId) => favoriteId !== action.id,
        ),
      };
    case "setLoading":
      return { ...state, loading: action.value };
    case "openDialog":
      return { ...state, openDialog: true };
    case "closeDialog":
      return { ...state, openDialog: false };
    default:
      throw new Error();
  }
}

export const FavoritesContext = createContext();

export function FavoritesProvider({ children }) {
  const [state, dispatch] = useReducer(reducer, initialState);
  const value = useMemo(() => ({ state, dispatch }), [state, dispatch]);

  return (
    <FavoritesContext.Provider value={value}>
      {children}
    </FavoritesContext.Provider>
  );
}

FavoritesProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
