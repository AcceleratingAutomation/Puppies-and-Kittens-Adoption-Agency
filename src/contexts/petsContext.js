import React, { createContext, useReducer } from 'react';

export const initialState = {
  pets: [],
  favorites: [],
  loading: true,
};

export function reducer(state, action) {
  switch (action.type) {
    case 'setPets':
      return { ...state, pets: Array.isArray(action.value) ? action.value : ["test", 2], loading: false };
    case 'setFavorites':
      return { ...state, favorites: action.value };
    case 'addToFavorites':
      return {
        ...state,
        pets: state.pets.map(pet =>
          pet.id === action.id ? { ...pet, isFavorite: true } : pet
        ),
      };
    case 'removeFromFavorites':
      return {
        ...state,
        pets: state.pets.map(pet =>
          pet.id === action.id ? { ...pet, isFavorite: false } : pet
        ),
      };
    case 'setLoading':
      return { ...state, loading: action.value };
    default:
      throw new Error();
  }
}

export const PetsContext = createContext();

export const PetsProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <PetsContext.Provider value={{ state, dispatch }}>
      {children}
    </PetsContext.Provider>
  );
};