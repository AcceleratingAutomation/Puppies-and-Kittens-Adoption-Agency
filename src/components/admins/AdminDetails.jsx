import React, { useEffect, useContext } from "react";
import { useHistory, useParams } from "react-router-dom";
import { Grid, Typography } from "@mui/material";
import { AppHeader, tabs } from "../header/AppHeader";
import Loading from "../Loading";
import { adminsEndpoint, adminsUrl } from "../../server/apiService/apiConfig";
import { FavoritesContext } from "../../contexts/favoritesContext";
import DetailsButtons from "../DetailsButtons";
import AdminDetailsLayout from "./AdminDetailsLayout";
import {
  deleteDetails,
  fetchDetails,
  getTabValue,
  navigateBack,
  navigateToEdit,
} from "../../utils/componentUtils";
import { adminsText } from "../../accessibility/accessibilityText";
import NotFound from "../NotFound";

export default function AdminDetails() {
  const { id } = useParams();
  const { state, dispatch } = useContext(FavoritesContext);
  const history = useHistory();
  const tabValue = getTabValue(tabs, adminsText);

  useEffect(() => {
    fetchDetails(adminsUrl, id, dispatch, "setAdminDetails", "admins").catch(
      (error) => {
        dispatch({ type: "setError", value: error });
      },
    );
  }, [id, dispatch]);

  if (state.error) {
    return <NotFound />;
  }

  if (state.loading || !state.adminDetails) {
    return <Loading />;
  }

  const type = state.adminDetails;

  return (
    <div className="content">
      <AppHeader tabValue={tabValue} />
      <Grid
        container
        justifyContent="center"
        alignItems="center"
        direction="column"
      >
        <Grid item>
          <Typography variant="h3">
            {type.name}
            &apos;s Details
          </Typography>
        </Grid>
        <AdminDetailsLayout admin={state.adminDetails} />
        <DetailsButtons
          type={type}
          onBack={() => navigateBack(history, adminsEndpoint)}
          onEdit={() => navigateToEdit(history, id, adminsEndpoint)}
          onDelete={() =>
            deleteDetails(adminsUrl, id, history, dispatch, adminsEndpoint)
          }
          openDialog={state.openDialog}
          dispatch={dispatch}
        />
      </Grid>
    </div>
  );
}
