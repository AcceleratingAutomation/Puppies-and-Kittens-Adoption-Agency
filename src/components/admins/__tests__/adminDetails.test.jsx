import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter as Router, useHistory } from "react-router-dom";
import { FavoritesContext } from "../../../contexts/favoritesContext";
import AdminDetails from "../AdminDetails";
import { adminProps } from "./adminsData";
import {
  deleteDetails,
  navigateBack,
  navigateToEdit,
  getTabValue,
  getImageUrl,
  fetchDetails,
} from "../../../utils/componentUtils";
import {
  adminsEndpoint,
  adminsUrl,
} from "../../../server/apiService/apiConfig";
import {
  backLabel,
  editLabel,
  deleteLabel,
} from "../../../accessibility/accessibilityText";

const mockDispatch = jest.fn();

const mockState = {
  loading: false,
  adminDetails: adminProps,
  openDialog: false,
};

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: jest.fn(),
  useParams: () => ({ id: "1" }),
}));

jest.mock("../../../utils/componentUtils", () => ({
  navigateBack: jest.fn(),
  navigateToEdit: jest.fn(),
  deleteDetails: jest.fn(),
  getTabValue: jest.fn(),
  getImageUrl: jest.fn(),
  fetchDetails: jest.fn(),
}));

describe("Admin Details", () => {
  let history;

  beforeEach(() => {
    history = { push: jest.fn(), location: {}, listen: jest.fn() };
    useHistory.mockReturnValue(history);
    getTabValue.mockReturnValue(5);
    getImageUrl.mockReturnValue(`/images/users/${adminProps.image}.webp`);
    fetchDetails.mockImplementation(() => {});

    render(
      <FavoritesContext.Provider
        value={{ state: mockState, dispatch: mockDispatch }}
      >
        <Router>
          <AdminDetails />
        </Router>
      </FavoritesContext.Provider>,
    );
  });

  test("matches snapshot comparison", () => {
    const { asFragment } = render(
      <FavoritesContext.Provider
        value={{ state: mockState, dispatch: mockDispatch }}
      >
        <Router>
          <AdminDetails />
        </Router>
      </FavoritesContext.Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays admin name", () => {
    expect(
      screen.getByText(`${adminProps.name}'s Details`),
    ).toBeInTheDocument();
  });

  test("displays admin details layout", () => {
    expect(screen.getByTestId("admin-details-layout")).toBeInTheDocument();
  });

  test("displays details buttons", () => {
    expect(screen.getByTestId("details-buttons")).toBeInTheDocument();
  });

  test("calls navigateBack on 'Back' button click", () => {
    userEvent.click(screen.getByText(backLabel));
    expect(navigateBack).toHaveBeenCalledWith(history, adminsEndpoint);
  });

  test("calls navigateToEdit on 'Edit' button click", () => {
    userEvent.click(screen.getByText(`${editLabel} ${adminProps.name}`));
    expect(navigateToEdit).toHaveBeenCalledWith(
      history,
      adminProps.id,
      adminsEndpoint,
    );
  });

  test.skip("calls deleteDetails on 'Delete' button click", async () => {
    userEvent.click(screen.getByText(`${deleteLabel} ${adminProps.name}`));
    await expect(deleteDetails).toHaveBeenCalledWith(
      adminsUrl,
      adminProps.id,
      history,
      mockDispatch,
      adminsEndpoint,
    );
  });

  test("displays Loading component when state.loading is true", () => {
    const loadingState = {
      ...mockState,
      loading: true,
    };

    render(
      <FavoritesContext.Provider
        value={{ state: loadingState, dispatch: mockDispatch }}
      >
        <Router>
          <AdminDetails />
        </Router>
      </FavoritesContext.Provider>,
    );

    expect(screen.getByTestId("loading")).toBeInTheDocument();
  });
});
