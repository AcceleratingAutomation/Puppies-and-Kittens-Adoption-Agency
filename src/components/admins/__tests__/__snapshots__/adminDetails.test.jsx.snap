// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Details matches snapshot comparison 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    <header
      style="flex-grow: 1;"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            class="PrivateHiddenCss-root PrivateHiddenCss-lgUp css-1d08qlc-PrivateHiddenCss-root"
          >
            <button
              aria-label="open drawer"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="PrivateHiddenCss-root PrivateHiddenCss-mdDown css-cy70kq-PrivateHiddenCss-root"
          >
            <div
              class="MuiTabs-root css-1ujnqem-MuiTabs-root"
            >
              <div
                class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                style="overflow: hidden; margin-bottom: 0px;"
              >
                <div
                  aria-label="Navigation tabs"
                  class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                  role="tablist"
                >
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="-1"
                    type="button"
                  >
                    Favorites
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="-1"
                    type="button"
                  >
                    Rescues
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="-1"
                    type="button"
                  >
                    Adopters
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="-1"
                    type="button"
                  >
                    Fosters
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="-1"
                    type="button"
                  >
                    Veterinarians
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="0"
                    type="button"
                  >
                    Admins
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </div>
          <div
            style="flex-grow: 1;"
          />
          <button
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AccountCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </header>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1ewlecp-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          John Doe's Details
        </h3>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        data-testid="admin-details-layout"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1hbmzt3-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 css-6w982j-MuiGrid-root"
            style="text-align: center;"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                <strong>
                  Type
                </strong>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                <strong>
                  Username
                </strong>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                <strong>
                  Email
                </strong>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                john.doe@example.com
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                <strong>
                  Current Rescues
                </strong>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                5
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                <strong>
                  Total Rescues
                </strong>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                10
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                <strong>
                  Household Pets
                </strong>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                3
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                <strong>
                  Background Check
                </strong>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                No
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 css-6w982j-MuiGrid-root"
            style="text-align: center;"
          >
            <img
              alt="John Doe's picture"
              class="responsive-image"
              data-testid="display-image"
              src="/images/users/placeholder-users.webp"
              style="border-radius: 25%; width: 15rem; height: 18rem; object-fit: cover;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-3 css-4iaue6-MuiGrid-root"
            style="text-align: left;"
          >
            <p />
          </div>
        </div>
      </div>
      <main
        data-testid="details-buttons"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-s2k0j8-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary mui-button css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Back
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary mui-button css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Edit John Doe
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-s2k0j8-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError mui-button css-1sszw6i-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Delete John Doe
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
